swarm/swarm.go

```diff
@@ -171,6 +173,10 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 	delivery := stream.NewDelivery(to, self.netStore)
 	self.netStore.NewNetFetcherFunc = network.NewFetcherFactory(delivery.RequestFromPeers, config.DeliverySkipCheck).New
 
+	if config.SwapEnabled {
+		self.swap = swap.New(stateStore)
```

Comment:
are you sure we can use the same stateStore for balances and intervals (below as an arg to `NewRegistry`)?
you can make sure they dont interfere by prefixing the peerID with 'balance' or something.
Or alternatively, you can create a separate state store called `balances.db`:

Reply:
Right, I had thought about this and in some instance I had it actually prefixed (probably in the second iteration implementation with the p2p events).

Actually, as we are writing to the `stateStore` on every balance change, I'd probably prefer a separate `balance.db`, prefixing seems a bit too much overhead?

Reply:
Will do with separate state store.